src/deriver/api.py

```diff
@@ -455,12 +426,18 @@ def derive_selfies_gb(self, n_children: int = 100, mut_rate: float = 0.01):
                 if self.data.filter_molecules:
                     if child not in self.data.filter_molecules:
                         good_children.append(child)
-                        self.data.all_good_selfies_gb_children.append(child)
+                        if kind == 'smiles':
+                            self.data.all_good_smiles_gb_children.append(child)
+                        else:
+                            self.data.all_good_selfies_gb_children.append(child)
                     else:
                         logger.debug(f"skipping previously seen molecule: {child}")
                 else:
                     good_children.append(child)
-                    self.data.all_good_selfies_gb_children.append(child)
+                    if kind == 'smiles':
+                        self.data.all_good_smiles_gb_children.append(child)
```

Comment:
since we're here doing edits, might as well mention: it's probably better to just keep appending to `good_children`, and then at the end just set `self.data.all_good_smiles_gb_children = good_children`